datafusion/physical-plan/src/windows/mod.rs

```diff
@@ -283,7 +283,11 @@ fn create_built_in_window_expr(
                 args[1]
                     .as_any()
                     .downcast_ref::<Literal>()
-                    .unwrap()
+                    .ok_or_else(|| {
+                        DataFusionError::Execution(
+                            format!("Expected a signed integer literal for the second argument of nth_value, got {}", args[1]),
+                        )
```

Comment:
```suggestion
                        exec_datafusion_err!("Expected a signed integer literal for the second argument of nth_value, got {}", args[1])
```

avoids the format call and wires up the backtrace. You may need to import it from here though:

https://github.com/apache/datafusion/blob/37e54ee874e9027a329ba4f6c1e0e6359d63a33c/datafusion/common/src/error.rs#L545-L546

PS: I should have caught that during the first round of reviews, sorry.